### ISSUE TITLE
Performance degradation of `DatabaseCleaner.strategy=` in database_cleaner >= v1.8.0

### ISSUE THREAD

**Issue by benedikt** [DatabaseCleaner/database_cleaner#624]
With the recent refactoring of database_cleaner into multiple gems, the performance of `DatabaseCleaner.strategy=` took a big hit. As it is not unusual that the method will be called in a `before(:each)` hook, this can significantly slow down a test suite. (In our case it's about 30x slower)

A simple benchmark outlines the problem:

```ruby
# Database Cleaner 1.7.0
puts Benchmark.measure { 1000.times { DatabaseCleaner.strategy = :transaction } }
# => 0.006961   0.002179   0.009140 (  0.010197)

# Database Cleaner 1.8.0
puts Benchmark.measure { 1000.times { DatabaseCleaner.strategy = :transaction } }
# => 31.013500  26.446908  57.460408 ( 57.592854)
```

I haven't fully figured out the root cause, but I suspect it has something to do with how the strategies are required via RubyGems instead of just the file system.

**Comment by benedikt** [comment]
I think it boils down to the [`DatabaseCleaner::Base#orm_module`](https://github.com/DatabaseCleaner/database_cleaner/blob/0f4e29d0adf207e961e1ea18562a828a8c1017fd/lib/database_cleaner/base.rb#L117-L123) method. It modifies the load path and attempts a new require every time it is called. This happens every time the strategy is set (`#strategy=` -> `#create_strategy` -> `#orm_strategy` -> `#orm_module`)

**Comment by hwo411** [comment]
I can confirm the performance degradation. After the recent gem upgrade of our app, tests started taking 2-2.5x more time, even if there is no database activity in the test suite at all. I was just going to report the problem and discovered this new issue.

@benedikt, thanks for working on it!

**Comment by botandrose** [comment]
Closed via #627
